### ISSUE TITLE
🚀 Feature: Catalog API to support searching

### ISSUE THREAD

**Issue by drodil** [backstage/backstage#15391]
### 🔖 Feature description

Catalog API GET /entities should allow finding components with wildcard search that would match at minimum the name of the components. 

### 🎤 Context

Using autocomplete to select components in a plugin now has to fetch all components. If there was a way to find entities with a string search parameter, this could be done asynchronously during typing like for example here: https://mui.com/material-ui/react-autocomplete/#search-as-you-type

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by drodil** [comment]
Yes that might be an option but I have use case where I would like to save the uid of the component as a relation to a question: https://github.com/drodil/backstage-plugin-qeta/issues/12 

Not sure if the uid of components is returned from search (at least by default?). Might be that I just have to go with loading all catalog items at load.

**Comment by freben** [comment]
Oh then the ask is actually moot - we don't use the uid at all, in basically any context. What you use to refer to an entity, is universally (this goes _everywhere_ in Backstage) their [entity ref](https://backstage.io/docs/features/software-catalog/references#string-references). So what both internal plugins and third party systems store, are strings like `component:default/foo`, not uids.

**Comment by freben** [comment]
Made #15394 to address that anyway. Thanks for mentioning it!

**Comment by drodil** [comment]
Thanks, @freben, for clearing this out before I went in the wrong direction 👍
